modules/saml/lib/IdP/SAML2.php

```diff
@@ -238,6 +238,31 @@ private static function getAssertionConsumerService(
         return $spMetadata->getDefaultEndpoint('AssertionConsumerService', $supportedBindings);
     }
 
+    /**
+     * Check a SOAP AuthnRequest.
+     *
+     * @param SimpleSAML_Configuration $idpMetadata The metadata for the IdP.
+     * @param SimpleSAML_Configuration $spMetadata  The metadata for the SP.
+     * @param array                    &$state      The state array.
+     */
+    private static function processSOAPAuthnRequest(SimpleSAML_Configuration $idpMetadata, SimpleSAML_Configuration $spMetadata, SAML2_AuthnRequest $request, array &$state)
+    {
+
+        // Send the response via SOAP.
+        $state['saml:Binding'] = \SAML2\Constants::BINDING_SOAP;
+
+        if (!sspmod_saml_Message::checkSign($spMetadata, $request)) {
+            throw new SimpleSAML_Error_Exception('SOAP authentication request not signed.');
```

Comment:
I haven't tested it yet, but it looks like replacing this block with:

```
sspmod_saml_Message::validateMessage($idpMetadata, $spMetadata, $request);
```

Should do the trick... but it looks like this is already performed on line 359 of `receiveAuthnRequest()`. I need to verify that's true before removing this.